spyder/plugins/framesexplorer/widgets/framesbrowser.py

```diff
@@ -0,0 +1,444 @@
+
+    def paint(self, painter, option, index):
+        """Paint the item."""
+        options = QStyleOptionViewItem(option)
+        self.initStyleOption(options, index)
+
+        style = (QApplication.style() if options.widget is None
+                 else options.widget.style())
+
+        doc = QTextDocument()
+        text = options.text
+        doc.setHtml(text)
+        doc.setDocumentMargin(0)
+
+        # This needs to be an empty string to avoid the overlapping the
+        # normal text of the QTreeWidgetItem
+        options.text = ""
+        style.drawControl(QStyle.CE_ItemViewItem, options, painter)
+
+        ctx = QAbstractTextDocumentLayout.PaintContext()
+
+        textRect = style.subElementRect(QStyle.SE_ItemViewItemText,
+                                        options, None)
+        painter.save()
+
+        painter.translate(textRect.topLeft())
+        painter.setClipRect(textRect.translated(-textRect.topLeft()))
+        doc.documentLayout().draw(painter, ctx)
+        painter.restore()
+
+    def sizeHint(self, option, index):
+        """Get a size hint."""
+        options = QStyleOptionViewItem(option)
+        self.initStyleOption(options, index)
+        doc = QTextDocument()
+        doc.setHtml(options.text)
+        doc.setTextWidth(options.rect.width())
+        size = QSize(int(doc.idealWidth()), int(doc.size().height()))
+        return size
+
+
+class ResultsBrowser(QTreeWidget):
+    sig_edit_goto = Signal(str, int, str)
+    sig_activated = Signal(int)
+    sig_show_namespace = Signal(dict)
+
+    def __init__(self, parent, color_scheme):
+        super().__init__(parent)
+        self.font = get_font()
+        self.data = None
+        self.threads = None
+        self.color_scheme = color_scheme
+        self.text_color = color_scheme['normal'][0]
+        self.frames = None
+        self.menu = None
+        self.empty_ws_menu = None
+        self.view_locals_action = None
+
+        # Setup
+        self.setItemsExpandable(True)
+        self.setColumnCount(1)
+        self.set_title('')
+        self.setSortingEnabled(False)
+        self.setItemDelegate(ItemDelegate(self))
+        self.setUniformRowHeights(True)  # Needed for performance
+        self.sortByColumn(0, Qt.AscendingOrder)
+
+        # Signals
+        self.header().sectionClicked.connect(self.sort_section)
+        self.itemActivated.connect(self.activated)
+        self.itemClicked.connect(self.activated)
+
+    def set_title(self, title):
+        self.setHeaderLabels([title])
+
+    def activated(self, item):
+        """Double-click event."""
+        itemdata = self.data.get(id(self.currentItem()))
+        if itemdata is not None:
+            filename, lineno = itemdata
+            self.sig_edit_goto.emit(filename, lineno, '')
+            # Index exists if the item is in self.data
+            self.sig_activated.emit(self.currentItem().index)
+        if self.parent().get_conf("show_locals_on_click"):
```

Comment:
Please avoid this by making `ResultsBrowser` inherit from the `SpyderConfigurationAccessor` mixin, which is in `spyder.api.config.mixins`. That adds the `get_conf`/`set_conf` methods to any Python class.